test-packages/test-utils/src/utils/semantic-model-builder.ts

```diff
@@ -100,7 +100,9 @@ export function buildUI5Association(
   };
 }
 
-export function buildUI5Class(opts: PartialWithName<UI5Class>): UI5Class {
+export function buildUI5Class<T extends PartialWithName<UI5Class>>(
+  opts: T
+): UI5Class & T {
```

Comment:
This is great for managing the optional vs known to exist props, should it be adjusted for all the builder functions?

Reply:
Hmm, but what if we pass something that is not a UI5Class prop?
Should T be constrained to properties that exist in UI5Class?

Reply:
I think so but we can do it when we need it (which we will for the other deprecated validations)

Reply:
> Hmm, but what if we pass something that is not a UI5Class prop?
> Should T be constrained to properties that exist in UI5Class?

The type definition will still be correct because it will contain all the sent properties.
I wouldn't do additional effort in this definition for test functions.

Reply:
> I wouldn't invest additional effort in this definition for test functions.

That is valid, but should the test functions at least be **consistent**?
When someone copies and creates a new such utility would they copy from the Class pattern or one of the other patterns?